service/dap/server_test.go

```diff
@@ -1710,10 +1710,17 @@ func substitutePathTestHelper(t *testing.T, fixture protest.Fixture, client *dap
 	}
 
 	launchAttachConfig["stopOnEntry"] = false
+	// All paths used in the 'substitutePath' attribute must be absolute directory paths.
```

Comment:
What happens if they are not? Should launch request handler catch this? Will this otherwise fail to map in subsequent stacktrace, breakpoint, etc requests that apply the mapping? Will this fail inside of Debugger command code? Have you given this a try already or this is another one of these things that we need to try and see (like the launch request program)?

Reply:
I don't think this requirement exists (but I could be wrong). Paths in the executable can be relative because of `-trimpath`.

Reply:
Correct I think it can be relative, as long as all of the other paths that we use are also relative. The dap server does not currently have support for relative paths or trimmed paths which is why I included this comment. The substitute path rules are applied to the beginning of the path, so they would have to have the same part trimmed.

Additionally the current code requires the paths to be directories, since a separator is added to the end of the string if it is not already there. So if you have a file /dir/main.go, and you have a mapping "from": /dir/main.go, "to": "/other/dir/main.go", it will not apply because "/dir/main.go/" is not a prefix of "/dir/main.go".

We could add some code to catch the paths early by checking if they have certain file extensions, but since this could be remote, the paths may not exist on this machine to check.